raft-tech/TANF-app
# Issue 1534: Fix zap scan parameters

## Summary of Changes
_Provide a brief summary of changes_

- updates zap version to latest available
- increases wait time for zap scan ci job to 60min
- updates url exclusion list s.t. only TDP URLs are scanned/reported. 

Pull request closes #1534 

## How to Test
_List the steps to test the PR_
- Check [here](https://app.circleci.com/pipelines/github/raft-tech/TANF-app?branch=feat%2F1534-updating-to-latest-zaproxy&filter=all) for the latest nightly backend scan (runs every hour at 10 after and 40 after until merge) and confirm that: (1) backend owasp scan did not timeout after 10 min and (2) only TDP sites are included in the results:
  - backend scan timing out after 10min and TDP sites included in the results (for successful jobs) (before)
![1534_33p5](https://user-images.githubusercontent.com/63075587/156649637-373f572a-24bc-40e8-84bb-4213d101999f.PNG)
![1534_33p3](https://user-images.githubusercontent.com/63075587/156649034-68de2d78-5d5d-4f88-81e8-e95b288f2232.PNG)

  - backend scan **_not_** timing out after 10min and non-TDP sites excluded in the results  (after)
 ![1534_33p2](https://user-images.githubusercontent.com/63075587/156649140-7daf8a66-a0b5-4b74-ae64-6749582c0d84.PNG)
![1534_33](https://user-images.githubusercontent.com/63075587/156647663-d26001c2-2684-4dee-8a0d-9221fee1ded2.PNG)

## Tabletop


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] nightly-owasp-scan workflow scan runs without timing out at 10min
- [x] non-TDP URLs are not reported in the backend scan

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)


_n/a: zap scan parameter changes only_

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)
